src/search/pdbs/rcg.cc

```diff
@@ -0,0 +1,87 @@
+#include "rcg.h"
+
+#include "../option_parser.h"
+
+#include "../task_proxy.h"
+
+#include "../utils/countdown_timer.h"
+#include "../utils/logging.h"
+#include "../utils/rng.h"
+
+#include <algorithm>
+#include <unordered_set>
+
+using namespace std;
+
+namespace pdbs{
+static bool time_limit_reached(
+    const utils::CountdownTimer &timer, utils::Verbosity verbosity) {
+    if (timer.is_expired()) {
+        if (verbosity >= utils::Verbosity::NORMAL) {
+            utils::g_log << "time limit reached." << endl;
+        }
+        return true;
+    }
+    return false;
+}
+
+Pattern generate_pattern_rcg(
+    int max_pdb_size,
+    double max_time,
+    int goal_variable,
+    utils::Verbosity verbosity,
+    const shared_ptr<utils::RandomNumberGenerator> &rng,
+    const TaskProxy &task_proxy,
+    vector<vector<int>> &cg_neighbors) {
+    utils::CountdownTimer timer(max_time);
+    int current_var = goal_variable;
+    unordered_set<int> visited_vars;
+    visited_vars.insert(current_var);
+    int pdb_size = task_proxy.get_variables()[current_var].get_domain_size();
```

Comment:
`task_proxy.get_variables()` is used three times and could be stored in a local variable.